.clang-format

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: GPL-2.0
+#
+# clang-format configuration file. Intended for clang-format >= 4.
+#
+# For more information, see:
+#
+#   Documentation/process/clang-format.rst
+#   https://clang.llvm.org/docs/ClangFormat.html
+#   https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+#
+---
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlines: Left # Unknown to clang-format-4.0, this can be launched once only to fix existing code and commented out for CI
+AlignOperands: true
+AlignTrailingComments: true
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+BinPackArguments: true
+BinPackParameters: true
+BraceWrapping:
+  AfterClass: false
+  AfterControlStatement: false
+  AfterEnum: false
+  AfterFunction: true
+  AfterNamespace: true
+  AfterObjCDeclaration: false
+  AfterStruct: false
+  AfterUnion: false
+  #AfterExternBlock: false # Unknown to clang-format-5.0
+  BeforeCatch: false
+  BeforeElse: false
+  IndentBraces: false
+  #SplitEmptyFunction: true # Unknown to clang-format-4.0
+  #SplitEmptyRecord: true # Unknown to clang-format-4.0
+  #SplitEmptyNamespace: true # Unknown to clang-format-4.0
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeTernaryOperators: false
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit: 120 # modern screens easily support 120 chars, the 80 char limit is outdated
+ContinuationIndentWidth: 8
+DisableFormat: false
+ExperimentalAutoDetectBinPacking: false
+#IncludeBlocks: Preserve # Unknown to clang-format-5.0
+IncludeCategories:
+  - Regex: '.*'
+    Priority: 1
+IndentCaseLabels: false
+#IndentPPDirectives: None # Unknown to clang-format-5.0
+IndentWidth: 8
+IndentWrappedFunctionNames: false
+KeepEmptyLinesAtTheStartOfBlocks: false
+MacroBlockBegin: ''
+MacroBlockEnd: ''
+MaxEmptyLinesToKeep: 1
+
+# Taken from git's rules
+#PenaltyBreakAssignment: 10 # Unknown to clang-format-4.0
+PenaltyBreakBeforeFirstCallParameter: 30
+PenaltyBreakComment: 10
+PenaltyBreakFirstLessLess: 0
+PenaltyBreakString: 10
+PenaltyExcessCharacter: 100
+PenaltyReturnTypeOnItsOwnLine: 60
+
+PointerAlignment: Left
```

Comment:
What we are currently using is not a valid argument - current style is not really a good point of reference :P
One could argue that both left and right alignments are valid - https://www.stroustrup.com/bs_faq2.html#whitespace
Since we are not going for pure K&R style, I would prefer applying the pointer alignment which is emphasizing the type, rather then syntax, but this is strictly personal preference.

Reply:
My view probably doesn't carry much weight, but I agree with @gilles-peskine-arm on this.
I can see the argument that the `*` should be a part of the type, because semantically that's what it is. However, that's more of an issue with the design of C - as C in practice links the `*` to the name, not the type.

The gotcha mentioned above is a problem, and even in cases where comma-separated declaration is used correctly, it looks untidy (`int* p,* q`?). If left-aligned pointer `*`s were to be used, I think it would also be best to ban comma-separated declarations, and make sure all declarations take place in separate statements.

Overall, I think the problems with left-aligned `*`s outweigh the semantic neatness, especially since anyone with more than a modicum of C experience is probably already used to dealing with the right-aligned `*`.

Reply:
Alright, so far I'm in the minority ;)

Reply:
+1 from me for "Make that Right."